### PR TITLE
Add cert creation functions

### PR DESCRIPTION
**Pull request by antevens**

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -2667,6 +2667,63 @@ function manage_service {
...
+    assert test -n "${ca_cert_path}" && test -r "${ca_cert_path}"
+    assert test -n "${ca_key_path}" && test -r "${ca_key_path}"
+    if ! [ -e "${new_cert_path}" ] ; then
+        openssl req -new -keyout "${new_key_path}" -nodes -newkey "${tls_key_type}" -subj "${tls_common_cert_attrib}/CN=${common_name}" | \
+            openssl x509 -req -CAkey "${ca_key_path}" -CA "${ca_cert_path}" -days "${tls_valid_days}" -set_serial "${RANDOM}" -sha512 -out "${new_cert_path}"
+    fi
+}
+
+# Creates a self signed cert/key pair if a cert does not exist in the path.
+# Optionally consumes common_name variable and appends to CN attribute.
```
[the remark this answers is not in the document]
Comment (antevens) [reply shtdlib.sh]: Do you mean we should add a more detailed description of what a common name is in a certificate?

[the remark this answers is not in the document]
Comment (hrpatel) [reply shtdlib.sh]: No, fix the case to match what the function uses. Same as the other comments

[the remark this answers is not in the document]
Comment (antevens) [reply shtdlib.sh]: ah because it's in multiple places, got ya

File: shtdlib.sh [hunk 2]
```diff
@@ -2667,6 +2667,63 @@ function manage_service {
     return 1
 }
 
+tls_common_cert_attrib="${tls_common_cert_attrib:-/C=ZZ/ST=None/L=None/O=None/OU=None}"
+tls_valid_days=${tls_valid_days:-3650}
+tls_key_type="${tls_key_type:-rsa:4096}"
+
+# Creates a Certificate Authority if one does not exist in the CA cert path.
+# Requires two arguments, paths to the key and certificate and key files.
+# Optionally consumes common_name variable and appends to CN attribute.
```
[the remark this answers is not in the document]
Comment (antevens) [reply shtdlib.sh]: don

File: shtdlib.sh [hunk 3]
```diff
@@ -2688,6 +2688,63 @@ function manage_service {
...
+    assert test -n "${ca_key_path}" && test -r "${ca_key_path}"
+    if ! [ -e "${new_cert_path}" ] ; then
+        openssl req -new -keyout "${new_key_path}" -nodes -newkey "${tls_key_type}" -subj "${tls_common_cert_attrib}/CN=${common_name}" | \
+            openssl x509 -req -CAkey "${ca_key_path}" -CA "${ca_cert_path}" -days "${tls_valid_days}" -set_serial "${RANDOM}" -sha512 -out "${new_cert_path}"
+    fi
+}
+
+# Creates a self signed cert/key pair if a cert does not exist in the path.
+# Optionally consumes COMMON_NAME variable and appends to CN attribute.
+function tls_create_self_signed_cert {
```
Comment (hrpatel) [inline shtdlib.sh]: No description of function arguments here

**Pull request merged**